### PR TITLE
Theme material-nf3

### PR DESCRIPTION
**Pull request by bim9262**
[**Update material-nf**](https://github.com/greshake/i3status-rust/commit/2225b77edcf240551105cf8a91c2fa5ea0534465) 

The `nf-mdi-*` icons have been deprecated. This commit exchanges them
for their `nf-md` counterparts.

[**Add thermometer progression for material-nf**](https://github.com/greshake/i3status-rust/commit/11778c6638c26f94113fd3b626098aff7f7fcab2)

### CODE REVIEW

**Comment by ammgws** [comment]
In reference to this: https://github.com/ryanoasis/nerd-fonts/issues/1096 ?

What was the difference between nf-mdi and nf-md? I can't remember...

**Comment by bim9262** [comment]
Yes, I think that they moved the old mdi icons to a new region so they renamed them all to just md so the names wouldn't overlap as well. I will note that the new icons do render a bit smaller than the old ones.

**Comment by ammgws** [comment]
> I will note that the new icons do render a bit smaller than the old ones.

Apparently...
https://github.com/ryanoasis/nerd-fonts/issues/1061

I imagine getting a few bug reports from that alone..

File: files/icons/material-nf.toml [hunk 1]
```diff
@@ -21,91 +21,95 @@ backlight = [
...
+    "\U000f0082", # nf-md-battery_90
+    "\U000f0079", # nf-md-battery
 ]
-bell = "\uf599" # nf-mdi-bell
-bell-slash = "\uf59a" # nf-mdi-bell_off
-bluetooth = "\uf5ae" # nf-mdi-bluetooth
-calendar = "\uf5ec" # nf-mdi-calendar
-cogs = "\uf992" # nf-mdi-settings
+bell = "\U000f009c" # nf-md-bell_outline
+bell-slash = "\U000f009b" # nf-md-bell_off
```
Comment (MaxVerevkin) [inline files/icons/material-nf.toml:39]: Why outline? And why only for `bell` but not `bell-slash`?

Comment (bim9262) [reply files/icons/material-nf.toml:39]: It matched what had been done in fa. I think that it's easier to tell the difference when one is the outline and one is filled. (I did the same with the toggles)

Comment (MaxVerevkin) [reply files/icons/material-nf.toml:39]: Now these two icons are not perfectly aligned: 

https://user-images.githubusercontent.com/34583604/219849036-6fb130df-5d96-4dcf-a68d-7e9c63696bba.mp4

Comment (MaxVerevkin) [reply files/icons/material-nf.toml:39]: (does this mp4 work for anyone?)

Comment (ammgws) [reply files/icons/material-nf.toml:39]: > (does this mp4 work for anyone?)

image

Comment (MaxVerevkin) [reply files/icons/material-nf.toml:39]: It works in `mpv` :(

Comment (bim9262) [reply files/icons/material-nf.toml:39]: That's interesting, it doesn't move for me


https://user-images.githubusercontent.com/1033745/219849545-422d04b5-7354-4f3a-aa19-27dfc8fda973.mp4

Comment (MaxVerevkin) [reply files/icons/material-nf.toml:39]: image

image

Comment (MaxVerevkin) [reply files/icons/material-nf.toml:39]: It shifts, but less... Anyway, this is not a big deal, just a bit weird.

**Comment by Finii** [comment]
I will fix 1061 before v3.0.0 goes out.

**Comment by bim9262** [comment]
Perhaps this should become a new theme, `material-nf3`?

**Comment by ammgws** [comment]
So if nerd fonts v3.0.0 is going to not just deprecate but also remove the codepoints that we are currently using then we should probably create `material-nf3`.

**Comment by bim9262** [comment]
v2 does write glyphs outside of the PUA. As noted on https://github.com/ryanoasis/nerd-fonts/issues/365:

> It is the problem that font_patcher writes glyphs from Material into the range: U+F500 - U+FD46. This range overlaps areas that should not use for such purpose.

One of the last comments from @Finii:

> Yes, according to plan this will come (glyphs removed) with v3.0.0, this was just one necessary intermediate step.

So, if I understand correctly, some of the codepoints we use will in today's material-nf will be removed in v3.

**Comment by Finii** [comment]
@bim9262 

Citation from https://github.com/ryanoasis/nerd-fonts/releases

Screenshot 2023-02-23 at 08 41 01

**Comment by bim9262** [comment]
Thanks @Finii!!

@ammgws, I'll rename this `material-nf3`

**Comment by MaxVerevkin** [comment]
> we should probably create material-nf3

Is this a temporal solution until the size of new icons is fixed?

**Comment by bim9262** [comment]
Looks great @Finii, thanks!!

If anyone else wants to test and hasn't built a font before, this is what I did on arch:

```
sudo pacman -S fontforge
cd /tmp
git clone --depth 1 https://github.com/ryanoasis/nerd-fonts.git
cd nerd-fonts
./font-patcher -s -ext ttf -c -out ~/.local/share/fonts/ src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols\ 2048\ EM\ Nerd\ Font\ Complete\ Blank.sfd
```
Restart i3status-rust

**Comment by bim9262** [comment]
@MaxVerevkin, @ammgws, 

Anything that should be done before this is merged?

**Comment by Finii** [comment]
> If anyone else wants to test and hasn't built a font before, this is what I did on arch:
> 
> ```
> sudo pacman -S fontforge
> cd /tmp
> git clone --depth 1 https://github.com/ryanoasis/nerd-fonts.git
> cd nerd-fonts
> ./font-patcher -s -ext ttf -c -out ~/.local/share/fonts/ src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols\ 2048\ EM\ Nerd\ Font\ Complete\ Blank.sfd
> ```

Cloning the repo even with `depth 1` is rather time consuming. I would recomment using the [docker image](https://hub.docker.com/r/nerdfonts/patcher) (well, that needs some docker setup), or

```diff
 sudo pacman -S fontforge
 cd /tmp
-git clone --depth 1 https://github.com/ryanoasis/nerd-fonts.git
+mkdir nerd-fonts
 cd nerd-fonts
+curl -LO https://github.com/ryanoasis/nerd-fonts/raw/master/FontPatcher.zip 
+unzip FontPatcher.zip
 ./font-patcher -s -ext ttf -c -out ~/.local/share/fonts/ src/unpatched-fonts/NerdFontsSymbolsOnly/NerdFontsSymbols\ 2048\ EM\ Nerd\ Font\ Complete\ Blank.sfd
```

**Comment by ammgws** [comment]
> @MaxVerevkin, @ammgws,
> 
> Anything that should be done before this is merged?

I was thinking perhaps we should instead keep `material-nf` pointed towards the latest version, and instead introduce `material-nf2.3` as a legacy option instead.

**Comment by bim9262** [comment]
We may want to wait until the next nerd-fonts release so that we don't have complaints about the small icons. (I imagine most people haven't installed them from src)

**Comment by bim9262** [comment]
Just saw https://github.com/greshake/i3status-rust/blob/4e45727dbb078f080d636207089ba42ca3928955/files/icons/material-nf.toml#L73-L79. We might just end up with this commit done piecemeal. Do we want to keep `nf-md-*` out of `material-nf.toml` still? It feels like we'll muddle the line between `material-nf2.3.toml` and `material-nf.toml`.

**Comment by MaxVerevkin** [comment]
>  Do we want to keep nf-md-* out of material-nf.toml still?

If I understand it correctly, the reason we don't just switch to nf-md-* is icons' size. I didn't notice any scaling issues with the wifi icons, so I added them. If the scaling issues will be resolved, I see no point in splitting `material-nf` in two themes.

**Comment by ammgws** [comment]
The split was aimed at people using specific distros where they might have nerdfonts in the repo but not i3status-rust, and end up with the former lagging behind the latter which would be updated by the user. But I just checked the nerd-fonts repo and it seems like it's not even in Debian or Ubuntu repos (the usual suspects) so maybe we don't need to keep the legacy icon set?

**Comment by bim9262** [comment]
All of the codepoints we're using (both nf-mdi and nf-md) are available in 2.3, so as long as the major repos that have nerdfonts are at least that up to date it shouldn't be required to keep two icon sets around.

**Comment by ammgws** [comment]
Ok that makes things simpler then

**Comment by bim9262** [comment]
v3.0.0 has been released, congratulations @Finii!

**Comment by MaxVerevkin** [comment]
Yay, `Nerd Font Propo` makes spacing even without `icons_format` hacks :tada:

**Pull request merged**